### ISSUE TITLE
"runtime Flatpak runtime" updates showing under OS updates

### ISSUE THREAD

**Issue by cassidyjames** [elementary/appcenter#1403]
As of today running master on my elementary OS 6 install, I am getting two updates listed under Operating System Updates, both called "runtime Flatpak runtime". Hitting "Update" doesn't appear to do anything.

Running `flatpak update` outputs:

```
 1.     org.gnome.Platform.Locale    3.36      u     flathub    < 323.0 MB (partial)
 2.     org.gnome.Platform           3.36      u     flathub    < 324.5 MB
```

Performing the Flatpak update from Terminal indeed gets rid of the OS updates listed in AppCenter.

**Comment by cassidyjames** [comment]
As a note: this prevents Flatpak app updates on affected systems. We also are not getting any sort of error dialog, the install just fails silently and the updates remain showing up.

**Comment by janxkoci** [comment]
I see this also in Hera, before I could see which runtimes are getting updated, now they are all listed with the same generic "name".

Also, after the same recent update, flatpak updates started to show up in AppCenter even on a netbook where I never used Sideload and installed two flatpak apps just from the terminal. Not getting automatic updates worked great for me as the netbook has just 32GB eMMC drive, but now flatpaks show up mixed with other updates. It's quite annoying, since I explicitly didn't allow AppCenter to handle flatpaks on this machine.

**Comment by davidmhewitt** [comment]
> I see this also in Hera

Yes, we already know about this issue (which is why this issue report exists) and there is already a fix pending for it.

> Not getting automatic updates worked great for me 

AppCenter doesn't do automatic updates? You have to open it and click the update button.

> since I explicitly didn't allow AppCenter to handle flatpaks on this machine.

The fact that AppCenter couldn't update/manage packages that were installed systemwide before was a limitation of AppCenter, not that you had "disallowed" AppCenter to update them by installing them with the terminal. Flatpak packages are larger than traditional packages, yes, but you already have them installed. Updating them shouldn't cause much extra disk space utilisation, so I hope this isn't an issue.

However, if you want to handle your updates individually, then the terminal is probably better suited for this job. Please feel free to open any additional issue reports that you feel could improve this situation, but this is irrelevant discussion for this bug that will get lost when the bug is closed.

**Comment by janxkoci** [comment]
Sorry, I only seen elementary OS 6 mentioned in the description, so I thought I will give heads up about Hera too, in case it was not known.

The thing about AppCenter handling flatpaks is that when I first read the blog about Sideload, I got the impression that if I don't use Sideload, then AppCenter will not see or care about flatpaks. This was indeed true until the recent update, which puzzled me, since from the wording of the blog post I understood the mechanics differently.

I was holding the flatpak updates for two reasons, which in retrospect should have been handled differently.

1. App updates usually don't take much extra space, you are right about that. That is until the app upgrades a toolkit and pulls a new version of runtime with it. However I mostly kept things this way from the time I knew less about flatpak and runtimes, so it's not fault of anyone other than me.
2. I was holding my flatpak of LibreOffice on version 6.3 due to regressions in later versions. I use flatpak as it works better than deb from PPA, even with the same version number.

To summarize, I now use `flatpak mask --user` to hold just the updates to specific apps, instead of everything. I was worried about runtime updates, but now I know that they won't happen if the app doesn't require them (e.g. when the app is on hold). So I have learned my lesson and won't bother you with this anymore :smile:
